{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import requests\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "ip = '35.190.208.167'\n",
    "user = 'radchenko'\n",
    "filename = 'test.py'\n",
    "\n",
    "with open(filename, 'rb') as f:\n",
    "        resp = requests.post('http://{}:5000/{}'.format(ip, user),\n",
    "                             files={'file': f})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================= test session starts ==============================\n",
      "platform linux -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0 -- /home/d_bolkunov/anaconda3/envs/py36raw/bin/python\n",
      "cachedir: .pytest_cache\n",
      "rootdir: /home/d_bolkunov/contents, inifile:\n",
      "collecting ... collected 24 items\n",
      "\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_types xfail           [  4%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_sizes[1-2] xfail      [  8%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_sizes[0-2] xfail      [ 12%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_values[1-2-sum] FAILED [ 16%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_values[1-2-mean] FAILED [ 20%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_values[0-2-sum] FAILED [ 25%]\n",
      "tests/itertools/test_groupby.py::TestGroupby::test_values[0-2-mean] FAILED [ 29%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_types XPASS             [ 33%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_sizes[some-correct0-2] xfail [ 37%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_sizes[some-correct1-3] xfail [ 41%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_sizes[ititititit-correct2-3] xfail [ 45%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_sizes[ititititit-correct3-2] xfail [ 50%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_sizes[ititititit-correct4-4] xfail [ 54%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_values[some-correct0-2] FAILED [ 58%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_values[some-correct1-3] FAILED [ 62%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_values[ititititit-correct2-3] FAILED [ 66%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_values[ititititit-correct3-2] FAILED [ 70%]\n",
      "tests/itertools/test_ngrams.py::TestNgrams::test_values[ititititit-correct4-4] FAILED [ 75%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[2] FAILED [ 79%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[3] FAILED [ 83%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[4] FAILED [ 87%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[5] FAILED [ 91%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[6] FAILED [ 95%]\n",
      "tests/itertools/test_permutations.py::TestPermutations::test_values[7] FAILED [100%]\n",
      "\n",
      "=================================== FAILURES ===================================\n",
      "_______________________ TestGroupby.test_values[1-2-sum] _______________________\n",
      "\n",
      "self = <tests.itertools.test_groupby.TestGroupby object at 0x7fda7e85ecf8>\n",
      "package_path = 'files/radchenko/test.py', group_col = 1, value_col = 2\n",
      "agg_function = <built-in function sum>\n",
      "\n",
      "    @pytest.mark.parametrize(\"agg_function\", [sum, mean], ids=['sum', 'mean'])\n",
      "    @pytest.mark.parametrize(\"group_col,value_col\", [(1, 2), (0, 2)])\n",
      "    def test_values(self, package_path, group_col, value_col, agg_function):\n",
      "        package = load_package(package_path)\n",
      "        correct = column_group_res[group_col, agg_function]\n",
      ">       out = package.group_by(DATA, group_col, value_col, agg_function)\n",
      "\n",
      "tests/itertools/test_groupby.py:71: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "data = [('Factory A', 'day 1', 100), ('Factory B', 'day 1', 200), ('Factory C', 'day 1', 100), ('Factory E', 'day 1', 300), ('Factory A', 'day 2', 200), ('Factory B', 'day 2', 200), ...]\n",
      "group_col = 1, value_col = 2, agg_function = <built-in function sum>\n",
      "\n",
      "    def group_by(data, group_col, value_col, agg_function):\n",
      "        \"\"\" Group by\n",
      "            Inputs:\n",
      "              - data: list of tuples e.g.: [('A', 2, 3), ('A', 3, 4)]\n",
      "              - group_col: int, column to group by e.g.: 0\n",
      "              - value_col: int, column to aggregate e.g: 1\n",
      "              - agg_function: agg function, so e.g. : sum\n",
      "                it is applied to list of values in value_col\n",
      "                for the group\n",
      "                e.g. sum([1, 2, 3])\n",
      "            Output:\n",
      "              - out: dict group_name -> value {'A': 5}\n",
      "        \"\"\"\n",
      "        out = {}\n",
      "        for curr in data:\n",
      "          i = curr[group_col]\n",
      "          j = curr[value_col]\n",
      "          if i in out:\n",
      ">           out[i] = agg_function(out[i], j)\n",
      "E           TypeError: 'int' object is not iterable\n",
      "\n",
      "files/radchenko/test.py:24: TypeError\n",
      "______________________ TestGroupby.test_values[1-2-mean] _______________________\n",
      "\n",
      "self = <tests.itertools.test_groupby.TestGroupby object at 0x7fda7e817128>\n",
      "package_path = 'files/radchenko/test.py', group_col = 1, value_col = 2\n",
      "agg_function = <function mean at 0x7fda7ea9fb70>\n",
      "\n",
      "    @pytest.mark.parametrize(\"agg_function\", [sum, mean], ids=['sum', 'mean'])\n",
      "    @pytest.mark.parametrize(\"group_col,value_col\", [(1, 2), (0, 2)])\n",
      "    def test_values(self, package_path, group_col, value_col, agg_function):\n",
      "        package = load_package(package_path)\n",
      "        correct = column_group_res[group_col, agg_function]\n",
      ">       out = package.group_by(DATA, group_col, value_col, agg_function)\n",
      "\n",
      "tests/itertools/test_groupby.py:71: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "data = [('Factory A', 'day 1', 100), ('Factory B', 'day 1', 200), ('Factory C', 'day 1', 100), ('Factory E', 'day 1', 300), ('Factory A', 'day 2', 200), ('Factory B', 'day 2', 200), ...]\n",
      "group_col = 1, value_col = 2, agg_function = <function mean at 0x7fda7ea9fb70>\n",
      "\n",
      "    def group_by(data, group_col, value_col, agg_function):\n",
      "        \"\"\" Group by\n",
      "            Inputs:\n",
      "              - data: list of tuples e.g.: [('A', 2, 3), ('A', 3, 4)]\n",
      "              - group_col: int, column to group by e.g.: 0\n",
      "              - value_col: int, column to aggregate e.g: 1\n",
      "              - agg_function: agg function, so e.g. : sum\n",
      "                it is applied to list of values in value_col\n",
      "                for the group\n",
      "                e.g. sum([1, 2, 3])\n",
      "            Output:\n",
      "              - out: dict group_name -> value {'A': 5}\n",
      "        \"\"\"\n",
      "        out = {}\n",
      "        for curr in data:\n",
      "          i = curr[group_col]\n",
      "          j = curr[value_col]\n",
      "          if i in out:\n",
      ">           out[i] = agg_function(out[i], j)\n",
      "E           TypeError: mean() takes 1 positional argument but 2 were given\n",
      "\n",
      "files/radchenko/test.py:24: TypeError\n",
      "_______________________ TestGroupby.test_values[0-2-sum] _______________________\n",
      "\n",
      "self = <tests.itertools.test_groupby.TestGroupby object at 0x7fda7e829860>\n",
      "package_path = 'files/radchenko/test.py', group_col = 0, value_col = 2\n",
      "agg_function = <built-in function sum>\n",
      "\n",
      "    @pytest.mark.parametrize(\"agg_function\", [sum, mean], ids=['sum', 'mean'])\n",
      "    @pytest.mark.parametrize(\"group_col,value_col\", [(1, 2), (0, 2)])\n",
      "    def test_values(self, package_path, group_col, value_col, agg_function):\n",
      "        package = load_package(package_path)\n",
      "        correct = column_group_res[group_col, agg_function]\n",
      ">       out = package.group_by(DATA, group_col, value_col, agg_function)\n",
      "\n",
      "tests/itertools/test_groupby.py:71: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "data = [('Factory A', 'day 1', 100), ('Factory B', 'day 1', 200), ('Factory C', 'day 1', 100), ('Factory E', 'day 1', 300), ('Factory A', 'day 2', 200), ('Factory B', 'day 2', 200), ...]\n",
      "group_col = 0, value_col = 2, agg_function = <built-in function sum>\n",
      "\n",
      "    def group_by(data, group_col, value_col, agg_function):\n",
      "        \"\"\" Group by\n",
      "            Inputs:\n",
      "              - data: list of tuples e.g.: [('A', 2, 3), ('A', 3, 4)]\n",
      "              - group_col: int, column to group by e.g.: 0\n",
      "              - value_col: int, column to aggregate e.g: 1\n",
      "              - agg_function: agg function, so e.g. : sum\n",
      "                it is applied to list of values in value_col\n",
      "                for the group\n",
      "                e.g. sum([1, 2, 3])\n",
      "            Output:\n",
      "              - out: dict group_name -> value {'A': 5}\n",
      "        \"\"\"\n",
      "        out = {}\n",
      "        for curr in data:\n",
      "          i = curr[group_col]\n",
      "          j = curr[value_col]\n",
      "          if i in out:\n",
      ">           out[i] = agg_function(out[i], j)\n",
      "E           TypeError: 'int' object is not iterable\n",
      "\n",
      "files/radchenko/test.py:24: TypeError\n",
      "______________________ TestGroupby.test_values[0-2-mean] _______________________\n",
      "\n",
      "self = <tests.itertools.test_groupby.TestGroupby object at 0x7fda7e817400>\n",
      "package_path = 'files/radchenko/test.py', group_col = 0, value_col = 2\n",
      "agg_function = <function mean at 0x7fda7ea9fb70>\n",
      "\n",
      "    @pytest.mark.parametrize(\"agg_function\", [sum, mean], ids=['sum', 'mean'])\n",
      "    @pytest.mark.parametrize(\"group_col,value_col\", [(1, 2), (0, 2)])\n",
      "    def test_values(self, package_path, group_col, value_col, agg_function):\n",
      "        package = load_package(package_path)\n",
      "        correct = column_group_res[group_col, agg_function]\n",
      ">       out = package.group_by(DATA, group_col, value_col, agg_function)\n",
      "\n",
      "tests/itertools/test_groupby.py:71: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "data = [('Factory A', 'day 1', 100), ('Factory B', 'day 1', 200), ('Factory C', 'day 1', 100), ('Factory E', 'day 1', 300), ('Factory A', 'day 2', 200), ('Factory B', 'day 2', 200), ...]\n",
      "group_col = 0, value_col = 2, agg_function = <function mean at 0x7fda7ea9fb70>\n",
      "\n",
      "    def group_by(data, group_col, value_col, agg_function):\n",
      "        \"\"\" Group by\n",
      "            Inputs:\n",
      "              - data: list of tuples e.g.: [('A', 2, 3), ('A', 3, 4)]\n",
      "              - group_col: int, column to group by e.g.: 0\n",
      "              - value_col: int, column to aggregate e.g: 1\n",
      "              - agg_function: agg function, so e.g. : sum\n",
      "                it is applied to list of values in value_col\n",
      "                for the group\n",
      "                e.g. sum([1, 2, 3])\n",
      "            Output:\n",
      "              - out: dict group_name -> value {'A': 5}\n",
      "        \"\"\"\n",
      "        out = {}\n",
      "        for curr in data:\n",
      "          i = curr[group_col]\n",
      "          j = curr[value_col]\n",
      "          if i in out:\n",
      ">           out[i] = agg_function(out[i], j)\n",
      "E           TypeError: mean() takes 1 positional argument but 2 were given\n",
      "\n",
      "files/radchenko/test.py:24: TypeError\n",
      "___________________ TestNgrams.test_values[some-correct0-2] ____________________\n",
      "\n",
      "self = <tests.itertools.test_ngrams.TestNgrams object at 0x7fda7e85e438>\n",
      "package_path = 'files/radchenko/test.py', text = 'some', n = 2\n",
      "correct = Counter({'me': 1, 'om': 1, 'so': 1})\n",
      "\n",
      "    @pytest.mark.parametrize(\"text,correct,n\", DATA)\n",
      "    def test_values(self, package_path, text, n, correct):\n",
      "        package = load_package(package_path)\n",
      "        out = package.count_ngrams(text, n)\n",
      ">       assert out == correct\n",
      "E       AssertionError: assert Counter() == Counter({'me': 1, 'om': 1, 'so': 1})\n",
      "E         Right contains more items:\n",
      "E         {'me': 1, 'om': 1, 'so': 1}\n",
      "E         Full diff:\n",
      "E         - Counter()\n",
      "E         + Counter({'me': 1, 'om': 1, 'so': 1})\n",
      "\n",
      "tests/itertools/test_ngrams.py:50: AssertionError\n",
      "___________________ TestNgrams.test_values[some-correct1-3] ____________________\n",
      "\n",
      "self = <tests.itertools.test_ngrams.TestNgrams object at 0x7fda7e825710>\n",
      "package_path = 'files/radchenko/test.py', text = 'some', n = 3\n",
      "correct = Counter({'ome': 1, 'som': 1})\n",
      "\n",
      "    @pytest.mark.parametrize(\"text,correct,n\", DATA)\n",
      "    def test_values(self, package_path, text, n, correct):\n",
      "        package = load_package(package_path)\n",
      "        out = package.count_ngrams(text, n)\n",
      ">       assert out == correct\n",
      "E       AssertionError: assert Counter() == Counter({'ome': 1, 'som': 1})\n",
      "E         Right contains more items:\n",
      "E         {'ome': 1, 'som': 1}\n",
      "E         Full diff:\n",
      "E         - Counter()\n",
      "E         + Counter({'ome': 1, 'som': 1})\n",
      "\n",
      "tests/itertools/test_ngrams.py:50: AssertionError\n",
      "________________ TestNgrams.test_values[ititititit-correct2-3] _________________\n",
      "\n",
      "self = <tests.itertools.test_ngrams.TestNgrams object at 0x7fda7e7d9c50>\n",
      "package_path = 'files/radchenko/test.py', text = 'ititititit', n = 3\n",
      "correct = Counter({'iti': 4, 'tit': 4})\n",
      "\n",
      "    @pytest.mark.parametrize(\"text,correct,n\", DATA)\n",
      "    def test_values(self, package_path, text, n, correct):\n",
      "        package = load_package(package_path)\n",
      "        out = package.count_ngrams(text, n)\n",
      ">       assert out == correct\n",
      "E       AssertionError: assert Counter() == Counter({'iti': 4, 'tit': 4})\n",
      "E         Right contains more items:\n",
      "E         {'iti': 4, 'tit': 4}\n",
      "E         Full diff:\n",
      "E         - Counter()\n",
      "E         + Counter({'iti': 4, 'tit': 4})\n",
      "\n",
      "tests/itertools/test_ngrams.py:50: AssertionError\n",
      "________________ TestNgrams.test_values[ititititit-correct3-2] _________________\n",
      "\n",
      "self = <tests.itertools.test_ngrams.TestNgrams object at 0x7fda7e829ef0>\n",
      "package_path = 'files/radchenko/test.py', text = 'ititititit', n = 2\n",
      "correct = Counter({'it': 5, 'ti': 4})\n",
      "\n",
      "    @pytest.mark.parametrize(\"text,correct,n\", DATA)\n",
      "    def test_values(self, package_path, text, n, correct):\n",
      "        package = load_package(package_path)\n",
      "        out = package.count_ngrams(text, n)\n",
      ">       assert out == correct\n",
      "E       AssertionError: assert Counter() == Counter({'it': 5, 'ti': 4})\n",
      "E         Right contains more items:\n",
      "E         {'it': 5, 'ti': 4}\n",
      "E         Full diff:\n",
      "E         - Counter()\n",
      "E         + Counter({'it': 5, 'ti': 4})\n",
      "\n",
      "tests/itertools/test_ngrams.py:50: AssertionError\n",
      "________________ TestNgrams.test_values[ititititit-correct4-4] _________________\n",
      "\n",
      "self = <tests.itertools.test_ngrams.TestNgrams object at 0x7fda7e85ea58>\n",
      "package_path = 'files/radchenko/test.py', text = 'ititititit', n = 4\n",
      "correct = Counter({'itit': 4, 'titi': 3})\n",
      "\n",
      "    @pytest.mark.parametrize(\"text,correct,n\", DATA)\n",
      "    def test_values(self, package_path, text, n, correct):\n",
      "        package = load_package(package_path)\n",
      "        out = package.count_ngrams(text, n)\n",
      ">       assert out == correct\n",
      "E       AssertionError: assert Counter() == Counter({'itit': 4, 'titi': 3})\n",
      "E         Right contains more items:\n",
      "E         {'itit': 4, 'titi': 3}\n",
      "E         Full diff:\n",
      "E         - Counter()\n",
      "E         + Counter({'itit': 4, 'titi': 3})\n",
      "\n",
      "tests/itertools/test_ngrams.py:50: AssertionError\n",
      "_______________________ TestPermutations.test_values[2] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e8250b8>\n",
      "package_path = 'files/radchenko/test.py', n = 2\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 2\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "_______________________ TestPermutations.test_values[3] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e816860>\n",
      "package_path = 'files/radchenko/test.py', n = 3\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 3\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "_______________________ TestPermutations.test_values[4] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e822710>\n",
      "package_path = 'files/radchenko/test.py', n = 4\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 4\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "_______________________ TestPermutations.test_values[5] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e806160>\n",
      "package_path = 'files/radchenko/test.py', n = 5\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 5\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "_______________________ TestPermutations.test_values[6] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e8167b8>\n",
      "package_path = 'files/radchenko/test.py', n = 6\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 6\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "_______________________ TestPermutations.test_values[7] ________________________\n",
      "\n",
      "self = <tests.itertools.test_permutations.TestPermutations object at 0x7fda7e829940>\n",
      "package_path = 'files/radchenko/test.py', n = 7\n",
      "\n",
      "    @pytest.mark.parametrize(\"n\", range(2, 8))\n",
      "    def test_values(self, package_path, n):\n",
      "        package = load_package(package_path)\n",
      "        correct = CHECK[n]\n",
      ">       out = package.num_bad_permutations(n)\n",
      "\n",
      "tests/itertools/test_permutations.py:28: \n",
      "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n",
      "\n",
      "n = 7\n",
      "\n",
      "    def num_bad_permutations(n):\n",
      "        \"\"\"\n",
      "        Number of permutations of length n\n",
      "        where each element is not on its place\n",
      "        Inputs:\n",
      "          - n: int, length of permutation\n",
      "        Output:\n",
      "          - num_perm: int, number of such permutations\n",
      "        \"\"\"\n",
      "        c = 0\n",
      ">       for i in itertools.permutations(range(n)):\n",
      "E       NameError: name 'itertools' is not defined\n",
      "\n",
      "files/radchenko/test.py:40: NameError\n",
      "=============== 15 failed, 8 xfailed, 1 xpassed in 0.23 seconds ================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(json.loads(resp.text))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# echo -e $(curl -i -X POST -F \"file=@./submission.py\" http://35.190.208.167:5000/<surname>)\n",
    "\n",
    "# <surname> - поменять надо\n",
    "# а еще название файла можно поменять, который отправляешь"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
